### PR TITLE
Add support for running Censys queries via analytics

### PR DESCRIPTION
**Pull request by itsmvd**
This allows Yeti users to add Censys queries as indicators, which will then be ran through the Censys API every 24 hours, tagging the identified IP addresses in Yeti. This will allow for infrastructure tracking/hunting directly from within Yeti.

### CODE REVIEW

**Comment by itsmvd** [comment]
Related PR for the frontend: https://github.com/yeti-platform/yeti-feeds-frontend/pull/142

File: plugins/analytics/public/censys.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    censys_queries, _ = indicator.Query.filter({'query_type': indicator.QueryType.censys})
+
+    for query in censys_queries:
+        ip_addresses = self.query_censys(hosts_api, query.pattern)
+        for ip in ip_addresses:
+          ip_object = Observable.add_text(ip)
+          ip_object.tag(query.relevant_tags)
+          query.link_to(ip_object, 'censys', f'IP found with Censys query: {query.pattern}')
+
+  def query_censys(self, api, query: str) -> set[str]:
```
Comment (tomchop) [inline plugins/analytics/public/censys.py:32]:
```suggestion
  def query_censys(self, api: CensysHosts, query: str) -> set[str]:
```

File: plugins/analytics/public/censys.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      "name": "Censys",
+      "description": "Executes Censys queries (stored as indicators) and tags the returned IP addresses.",
+      "frequency": timedelta(hours=24),
+  }
+
+  def run(self):
+    hosts_api = CensysHosts(
+        api_id=yeti_config.get("censys", "api_key"),
+        api_secret=yeti_config.get("censys", "secret")
+        )
```
Comment (tomchop) [inline plugins/analytics/public/censys.py:22]: Please check that these two config settings are set and bail with a warning (logging.warning) if they're not?

File: plugins/analytics/public/censys.py
Comment (tomchop) [inline plugins/analytics/public/censys.py:1]: Please give it a ruff formatter pass (We'll add a github action check soon)

File: tests/analytics_test.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+from core.schemas import indicator
+from core.schemas.indicator import DiamondModel
+from core.schemas.observable import ObservableType
+from core.schemas import observable
+
+
+class AnalyticsTest(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls) -> None:
+        database_arango.db.connect(database="yeti-test")
```
Comment (tomchop) [inline tests/analytics_test.py:15]:
```suggestion
        database_arango.db.connect(database="yeti_test")
```

File: tests/analytics_test.py
Comment (tomchop) [inline tests/analytics_test.py:1]: Same comment as the other file, please give this a ruff formatter pass

**Review by tomchop — CHANGES_REQUESTED** [review]

**Review by tomchop — APPROVED** [review]
LGTM, thanks!!

**Pull request merged**